webui/src/pages/repositories/repository/fileRenderers/data.tsx

```diff
@@ -58,59 +88,54 @@ LIMIT 20`
                 <>
-                    <Table striped bordered hover size={"sm"} responsive={"sm"}>
-                        <thead>
-                        <tr>
-                            {Object.getOwnPropertyNames(res[0]).map(name =>
-                                <th key={name}>{name}</th>
-                            )}
-                        </tr>
-                        </thead>
-                        <tbody>
-                        {res.map((row, i) => (
-                            <tr key={`row-${i}`}>
-                                {Object.getOwnPropertyNames(res[0]).map(name => (
-                                    <DataRow key={`row-${i}-${name}`} value={row[name]}/>
-                                ))}
-                            </tr>
-                        ))}
-                        </tbody>
-                    </Table>
                     {(res.length < data.length) &&
-                        <Alert>{`Showing only the first ${res.length} rows (out of ${data.length})`}</Alert>
+                        <small>{`Showing only the first ${res.length.toLocaleString()} rows (out of ${data.length.toLocaleString()})`}</small>
                     }
+                    <div className="object-viewer-sql-results">
+                        <Table striped bordered hover size={"sm"} responsive={"sm"} className="sticky-table">
+                            <thead className="thead-dark">
+                            <tr>
+                                {Object.getOwnPropertyNames(res[0]).map(name =>
+                                    <th key={name}>{name}</th>
+                                )}
+                            </tr>
+                            </thead>
+                            <tbody>
+                            {res.map((row, i) => (
+                                <tr key={`row-${i}`}>
+                                    {Object.getOwnPropertyNames(res[0]).map(name => (
+                                        <DataRow key={`row-${i}-${name}`} value={row[name]}/>
+                                    ))}
+                                </tr>
+                            ))}
+                            </tbody>
+                        </Table>
+                    </div>
                 </>
             )
         }
     }
 
-
     return (
         <div>
-            <Form onSubmit={(e) => {
-                e.preventDefault()
-                setShouldSubmit(!shouldSubmit)
-            }}>
+            <Form onSubmit={handleSubmit}>
                 <Form.Group className="mb-2 mt-2" controlId="objectQuery">
-                    <Form.Control as="textarea" className="object-viewer-sql-input" rows={5} value={query} spellCheck={false} onChange={e => {
-                        setQuery(e.target.value)
-                    }} />
+                    <Form.Control as="textarea" className="object-viewer-sql-input" rows={5} defaultValue={initialQuery} spellCheck={false} ref={sql} />
```

Comment:
Not sure if it's something you want, but maybe add `autocomplete={false}`, so the native browser autocomplete thing doesn't pop up (unless you want. that, of course)

Reply:
Added!